packages/integrations/cloudflare/README.md

```diff
@@ -38,44 +38,45 @@ export default defineConfig({
 
 #### `mode:directory`
 
-Switching to directory mode allows you to use [pages plugins](https://developers.cloudflare.com/pages/platform/functions/plugins/) such as [Sentry](https://developers.cloudflare.com/pages/platform/functions/plugins/sentry/) or write custom code to enable logging.
+Switching to directory mode allows you to use [pages plugins](https://developers.cloudflare.com/pages/platform/functions/plugins/) or add custom functions manually.
 
-```ts
+```js
 // astro.config.mjs
 export default defineConfig({
   adapter: cloudflare({ mode: 'directory' }),
 });
 ```
 
-In `directory` mode, the adapter will compile the client-side part of your app the same way as in `advanced` mode by default, but moves the worker script into a `functions` folder in the project root. In this case, the adapter will only ever place a `[[path]].js` in that folder, allowing you to add additional plugins and pages middleware which can be checked into version control.
+In `directory` mode, the adapter will compile the client-side part of your app the same way as in `advanced` mode by default, but moves the worker script into the `functions` folder in the project root. In this case, the adapter will only ever place a `[[path]].js` in that folder, allowing you to add additional files manually.
 
 To instead compile a separate bundle for each page, set the `functionPerPath` option in your Cloudflare adapter config. This option requires some manual maintenance of the `functions` folder. Files emitted by Astro will overwrite existing `functions` files with identical names, so you must choose unique file names for each file you manually add. Additionally, the adapter will never empty the `functions` folder of outdated files, so you must clean up the folder manually when you remove pages.
 
-```diff
+```js ins={7-8}
+// astro.config.mjs
 import {defineConfig} from "astro/config";
 import cloudflare from '@astrojs/cloudflare';
 
 export default defineConfig({
      adapter: cloudflare({
         mode: 'directory',
-+       functionPerRoute: true
+        functionPerRoute: true
     })
 })
 ```
 
-Note that this adapter does not support using [Cloudflare Pages Middleware](https://developers.cloudflare.com/pages/platform/functions/middleware/). Astro will bundle the [Astro middleware](https://docs.astro.build/en/guides/middleware/) into each page.
+This adapter doesn't support [`edgeMiddleware`](https://docs.astro.build/en/reference/adapter-reference/#edgemiddleware). If you have a [Astro middleware](https://docs.astro.build/en/guides/middleware/) in your project and set the adapter configuration `functionPerRoute: true`, Astro will bundle the [Astro middleware](https://docs.astro.build/en/guides/middleware/) into each page bundle.
 
```

Comment:
Not happy with wording, but the technical aspect is correct. This will also be touched again, once _native_ [Cloudflare middlewares](https://developers.cloudflare.com/pages/platform/functions/middleware/) are fully supported.